runtimes/python-3.10/server.py

```diff
@@ -47,58 +47,97 @@ def redirect(self, url, statusCode = 301, headers = {}):
         return self.send('', statusCode, headers)
 
 class Request:
-    rawBody = None
+    bodyString = None
     body = None
     headers = None
     method = None
     url = None
+    path = None
+    port = None
+    query = None
+    queryString = None
+    scheme = None
+    host = None
 
 class Context:
     req = Request()
     res = Response()
 
-    _logs = []
-    _errors = []
+    logs = []
+    errors = []
 
     def __init__(self):
-        self._logs = []
-        self._errors = []
+        self.logs = []
+        self.errors = []
         self.req = Request()
         self.res = Response()
 
-    # TODO: Support for infinite parameters
-    # TODO: Support for objects (stringify)
     def log(self, message):
-        self._logs.append(str(message))
+        if isinstance(message, (list, dict, tuple)):
+            self.logs.append(json.dumps(message, separators=(',', ':')))
+        else:
+            self.logs.append(str(message))
 
     def error(self, message):
-        self._errors.append(str(message))
+        if isinstance(message, (list, dict, tuple)):
+            self.errors.append(json.dumps(message, separators=(',', ':')))
+        else:
+            self.errors.append(str(message))
 
 HTTP_METHODS = ['GET', 'HEAD', 'POST', 'PUT', 'DELETE', 'CONNECT', 'OPTIONS', 'TRACE', 'PATCH']
 
 @app.route('/', defaults={'u_path': ''}, methods = HTTP_METHODS)
 @app.route('/<path:u_path>', methods = HTTP_METHODS)
 def handler(u_path):
-    if (request.headers.get('x-open-runtimes-secret') != os.getenv('OPEN_RUNTIMES_SECRET')):
+    if (request.headers.get('x-open-runtimes-secret', '') != os.getenv('OPEN_RUNTIMES_SECRET', '')):
```

Comment:
This condition will be `false` when both the header and env var is empty.

In which case the security check is bypasses

Reply:
Added check for empty/missing secret header to always return 500.